allow for driver sources coming from agent-kmodule.tgz

iurly opened the pull request:
The current procedure is based on the assumption of having a full checkout of agent-libs.
However, the source code (for both kernel module and eBPF probe) is also available as the draios-agent-kmodule.tar.gz file where all parameters are already populated.
Enable that scenario while also retaining compatibility with the existing one.
Notice the unavoidable tweaks since there is no longer a driver/ directory available.

iurly committed:
allow for driver sources coming from agent-kmodule.tgz

The current procedure is based on the assumption of
having a full checkout of agent-libs.
However, the source code (for both kernel module and eBPF
probe) is also available as the draios-agent-kmodule.tar.gz
file where all parameters are already populated.
Enable that scenario while also retaining compatibility with
the existing one.
Notice the unavoidable tweaks since there is no longer
a driver/ directory available.

gnosek reviewed (dismissed):
nice!

deepskyblue86 commented on builder-entrypoint.sh:
```diff
@@ -23,6 +23,14 @@ ARCH=$(uname -m)
+	# Make sure we've been passed a full checkout of libs,
+	# (on which we're forced to be running cmake to configure)
+	# as opposed to an already-configured source package
+	if [[ ! -f ${SRC_DIR}/CMakeLists.txt ]]; then
```
Do we have any symlink case? If so, I'd better go with `-e`

iurly replied:
OK done!

nathan-b reviewed (dismissed):
:heart:

mavimo reviewed (commented):
@iurly have you already checked if we don't need to edit `builder-entrypoint-coreos.sh` as well?

iurly commented:
> @iurly have you already checked if we don't need to edit `builder-entrypoint-coreos.sh` as well?

Hmm... you're making a good point but it looks like that file is only used by `Dockerfile.coreos-old` which in turn seems completely unused. So maybe we could just remove them both? @gnosek can you please shed some light on this?

iurly committed:
address review comments

- use -e instead of -f when testing for CMakeLists.txt
- add/rewrite comments to hopefully clarify the behavior

iurly commented:
> @iurly have you already checked if we don't need to edit `builder-entrypoint-coreos.sh` as well?

CoreOS is long gone. Removing the remnants with https://github.com/draios/probe-builder/pull/100
